src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -476,6 +476,11 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
 
         manager.register_pass<ov::pass::CommonOptimizations>();
 
+        // In the case of "input -> reshape -> convert -> multiply",
+        // the "input -> reshape" subgraph is constant-folded in the above "CommonOptimizations"
+        // To handle this case, "KeepConstPrecision" is executed again.
+        manager.register_pass<ov::pass::KeepConstPrecision>(supported_woq_types, !device_info.supports_immad);
+
```

Comment:
Though it LGTM, I think we need a test case for this. Could you add such case in `src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/matmul_weights_decompression.cpp`?

Reply:
I'll add tests in the next PR. Thank you!